pkg/operator/resource/resourceapply/apps.go

```diff
@@ -15,13 +18,42 @@ import (
 	"github.com/openshift/library-go/pkg/operator/resource/resourcemerge"
 )
 
-// ApplyDeployment merges objectmeta and requires matching generation. It returns the final Object, whether any change as made, and an error
-func ApplyDeployment(client appsclientv1.DeploymentsGetter, recorder events.Recorder, required *appsv1.Deployment, expectedGeneration int64,
+const specHashAnnotation = "operator.openshift.io/spec-hash"
+
+// ApplyDeployment merges ObjectMeta of the provided deployment with an existing one
+// if it exists and updates the API if the deployment spec and metadata differ. To
```

Comment:
if the deployment spec or metadata differ from the previously required spec and metadata.  To be reliable, the input of the required spec from the operator should be stable.  It does not need to set all fields, since some fields are defaulted server-side.  

Detection of spec drift from intent by other actors is determined by generation, not by spec comparison.

Reply:
Done